### ISSUE TITLE
Go To Definition is broken on Step Definitions with named parameters that have restrictive types.

### ISSUE THREAD

**Issue by mrkaiser** [cucumber/language-service#123]
### 👓 What did you see?
On `go to definition` calls with feature files that have named parameters on `Scenario Outlines` , substitution does not happen. 
This is a bug a because in the documentation it is stated that replacement will happen. Furthermore this breaks the contract between the Cucumber expression's type when used directly  or with a examples. The types should not have to change depending on context. 

### ✅ What did you expect to see?

Some substitution should happen before the cucumber match check. Since the feature file location is in memory and we have access to the `cucumber/gherkin` library, it should be possible to at least do substitution with the first item in the examples table for a match. 

### 📦 Which tool/library version are you using?

This likely affects all languages, as I have verified it is broken in both python and ruby.

### 🔬 How could we reproduce it?

```
Scenario Outline: Anything
Given there are <n_cukes> cucumbers
```
A cucumber expression like so done in the language of your choice. Below if Python
```
@given("there are {float} cucumbers")
```
or in Ruby
```
Given("there are {float} cucumbers do |n_cukes|
End
````

This cucumber expression will only work if you overload and have an expression like so:
`there are {word} cucumbers` or `there are {} cucumbers`

Please outline the steps below:
-->

1. Build a feature file like the above
2. Build a type restrictive cucumber expression
3. Use the expression with named parameters and an examples table
4. Attempt a go to definition 

### 📚 Any additional context?

I am happy to help here, but I could not figure out how example substitution happens. A point in the right direction would be helpful.

**Comment by brasmusson** [comment]
In `cucumber/gherkin` the substitution is performed by the interpolate function in the compiler: https://github.com/cucumber/gherkin/blob/4f436c5a77a3f3c8a07830a9f3fa5213489b8193/javascript/src/pickles/compile.ts#L259

**Comment by kieran-ryan** [comment]
Closing as a duplicate of #149
